src/sdam/description/topology/mod.rs

```diff
@@ -701,6 +740,31 @@ impl SessionSupportStatus {
     }
 }
 
+/// Enum representing whether transactions are supported by the topology.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub(crate) enum TransactionSupportStatus {
+    /// It is not known yet whether the topology supports transactions. This is possible if no
+    /// data-bearing servers have updated the `TopologyDescription` yet.
+    Undetermined,
+
+    /// Transactions are not supported by this topology.
+    Unsupported,
+
+    /// Transactions are supported by this topology. A topology supports transactions if it
+    /// supports sessions and its maxWireVersion >= 7, or its maxWireVersion >= 8 and the topology
+    /// type is sharded.
+    ///
+    /// Note that meeting these conditions does not guarantee that a deployment
+    /// supports transactions; any other missing qualification will be reported by the server.
+    Supported,
```

Comment:
Can we update this docstring to indicate we don't support sharded transactions yet + TODO?